docs/csharp/codedoc.md

```diff
@@ -2,26 +2,1348 @@
+    Contains all methods for performing basic math functions
+*/
+/// <summary>
+/// The main Math class.
+/// Contains all methods for performing basic math functions.
+/// </summary>
+public class Math
+{
+    // Adds two integers and returns the result
+    /// <summary>
+    /// Adds two integers and returns the result.
+    /// </summary>
+    public static int Add(int a, int b)
+    {
+        // If any parameter is equal to the max value of an integer
+        // and the other is greater than zero
+        if ((a == int.MaxValue && b > 0) || (b == int.MaxValue && a > 0))
+            throw new System.OverflowException();
+
+        return a + b;
+    }
+}
+```
+
+The `<summary>` tag is super important and I strongly advice it be included because its content is the primary source of type or member description in the resulting API reference document. 
+
+### &lt;remarks&gt;
+
+The `<remarks>` tag is used to add information about types or members, supplementing the information specified with `<summary>`.
+In this example I'll just add it to the class.
+
+```csharp
+/*
+    The main Math class
+    Contains all methods for performing basic math functions
+*/
+/// <summary>
+/// The main Math class.
+/// Contains all methods for performing basic math functions.
+/// </summary>
+/// <remarks>
+/// This class can add, subtract, multiply and divide.
+/// </remarks>
+public class Math
+{
+
+}
+```
+
+### &lt;returns&gt;
+
+As the name suggests the `<returns>` tag is used in the comment for a method declaration to describe its return value.
+Like before this will be illustrated on the first `Add` method go ahead an implement it on other methods.
+
+```csharp
+// Adds two integers and returns the result
+    /// <summary>
+    /// Adds two integers and returns the result.
+    /// </summary>
+    /// <returns>
+    /// The sum of two integers.
+    /// </returns>
+    public static int Add(int a, int b)
+    {
+        // If any parameter is equal to the max value of an integer
+        // and the other is greater than zero
+        if ((a == int.MaxValue && b > 0) || (b == int.MaxValue && a > 0))
+            throw new System.OverflowException();
+
+        return a + b;
+    }
+```
+
+### &lt;value&gt;
+
+The `<value>` works similarly to the `<returns>` tag except that it is used for properties.
+Let's assume our `Math` library had a static property called `PI` here's how we'll use this tag:
+
+```csharp
+/*
+    The main Math class
+    Contains all methods for performing basic math functions
+*/
+/// <summary>
+/// The main Math class.
+/// Contains all methods for performing basic math functions.
+/// </summary>
+/// <remarks>
+/// This class can add, subtract, multiply and divide.
+/// </remarks>
+public class Math
+{
+    /// <value>Gets the value of PI.</value>
+    public static double PI { get; }
+}
+```
+
+### &lt;example&gt;
+
+The `<example>` tag lets you specify an example of how to use a class, method or other member. 
```

Comment:
This is written in passive voice. It will read stronger as an active voice statement: 

You use the `<example>` tag to include an example in your XML documentation.

(This is the second style comment on this article. It's also repeated several times, and I'll try to catch all of them.)